tests/cpp/unit-tests/test_client_ensemble.cpp

```diff
@@ -0,0 +1,210 @@
+#include "../../../third-party/catch/catch.hpp"
+#include "client.h"
+#include "dataset.h"
+#include "../client_test_utils.h"
+
+using namespace SmartRedis;
+
+// helper function for resetting environment
+// variables to their original state
+void reset_env_vars(const char* old_keyin, const char* old_keyout)
+{
+    if (old_keyin != nullptr) {
+        std::string reset_keyin =
+            std::string("SSKEYIN=") + std::string(old_keyin);
+        char* reset_keyin_c = new char[reset_keyin.size() + 1];
+        std::copy(reset_keyin.begin(), reset_keyin.end(), reset_keyin_c);
+        reset_keyin_c[reset_keyin.size()] = '\0';
+        putenv( reset_keyin_c);
+        delete [] reset_keyin_c;
+    }
+    else {
+        unsetenv("SSKEYIN");
+    }
+    if (old_keyout != nullptr) {
+        std::string reset_keyout =
+            std::string("SSKEYOUT=") + std::string(old_keyout);
+        char* reset_keyout_c = new char[reset_keyout.size() + 1];
+        std::copy(reset_keyout.begin(), reset_keyout.end(), reset_keyout_c);
+        reset_keyout_c[reset_keyout.size()] = '\0';
+        putenv( reset_keyout_c);
```

Comment:
As we discussed and found out together, `std::string::c_str()` returns `const char*` and `putenv` has the signature `int putenv(char*)`, so we will keep the code as is in order to avoid casting from `const char*` to `char*`